config/config.go

```diff
@@ -561,6 +586,7 @@ type P2PConfig struct { //nolint: maligned
 	// peers. If another node asks it for addresses, it responds and disconnects.
 	//
 	// Does not work if the peer-exchange reactor is disabled.
+	// automatically set true if Mode of the node is seednode
 	SeedMode bool `mapstructure:"seed-mode"`
```

Comment:
I think this depends on the p2p refactor. Did we decide to pull out the seed mode in a different binary or?

Reply:
I believe that is still up for discussion, but from the last conversations I heard, moving it to a separate binary is strong contender and there is a possibility that some part of this PR will have to adjusted for that.

Reply:
Hey @dongsam, we had a call yesterday and briefly discussed this, coming to the agreement that we would like to remove the `SeedMode` config parameter and rely only on the new `mode` parameter. This will help draw a clearer distinction between operating a seed node and a full node.

Reply:
As to the decision about having it as a separate binary, I'm not too familiar with what the advantages and disadvantages would be. Is it just so that is of a smaller size? 

If so, then I'm on board with this. However, I would also want to keep it as part of the main Tendermint binary as convenience to me when I am spinning up test nets and what not. Since we are down this road, is there any argument for the light client to also have its own binary?

Reply:
I think the idea is that it's the same binary, but it runs in either a completely isolated mode -- full-node, validator, or seed-mode, but not any combo. i.e. if seed-mode is enabled, it will only run in seed-mode without full-node functionality.

Reply:
thanks for the discussion and the feedback,  I'll make an additional commit to remove the `SeedMode` config parameter and rely only on the new `mode` parameter